website/src/pages/api/admin/users.ts

```diff
@@ -1,31 +1,44 @@
 import { withRole } from "src/lib/auth";
+import { oasstApiClient } from "src/lib/oasst_api_client";
 import prisma from "src/lib/prismadb";
 
-// The number of users to fetch in any request.
-const PAGE_SIZE = 20;
-
 /**
  * Returns a list of user results from the database when the requesting user is
  * a logged in admin.
  */
 const handler = withRole("admin", async (req, res) => {
-  // Figure out the pagination index and skip that number of users.
-  //
-  // Note: with Prisma this isn't the most efficient but it's the only possible
-  // option with cuid based User IDs.
-  const { pageIndex } = req.query;
-  const skip = parseInt(pageIndex as string) * PAGE_SIZE || 0;
+  // TODO(#673): Update this to support pagination.
+
+  // First, get all the users according to the backend.
+  const all_users = await oasstApiClient.fetch_users(20);
 
-  // Fetch 20 users.
-  const users = await prisma.user.findMany({
+  // Next, get all the users stored in the web's auth datbase to fetch their role.
+  const local_user_ids = all_users.map(({ id }) => id);
+  const local_users = await prisma.user.findMany({
+    where: {
+      id: {
+        in: local_user_ids,
+      },
+    },
     select: {
       id: true,
       role: true,
-      name: true,
-      email: true,
     },
-    skip,
-    take: PAGE_SIZE,
+  });
+
+  // Combine the information by updating the set of full users with their role.
+  // Default any users without a role set locally as "general".
+  const local_user_map = local_users.reduce((result, user) => {
+    result.set(user.id, user.role);
+    return result;
+  }, new Map());
+
+  const users = all_users.map((user) => {
+    const role = local_user_map.get(user.id) || "general";
+    return {
+      ...user,
+      role,
+    };
```

Comment:
will this happen every time you open the users dashboard?

Reply:
Yes, it's not ideal and not something I want to keep for long.  I think the real fix is to push the role information to the backend but that would require a variety of new routes to make that fetching and setting easier.